std/node/buffer.ts

```diff
@@ -247,7 +253,10 @@ export default class Buffer extends Uint8Array {
     sourceStart = 0,
     sourceEnd = this.length,
   ): number {
-    const sourceBuffer = this.subarray(sourceStart, sourceEnd);
+    const sourceBuffer = this
+      .subarray(sourceStart, sourceEnd)
+      .subarray(0, Math.max(0, targetBuffer.length - targetStart));
+
     targetBuffer.set(sourceBuffer, targetStart);
```

Comment:
throws with `Buffer.alloc(5).copy(Buffer.alloc(0), 5)`

```suggestion
    if (sourceBuffer.length === 0) return 0;
    targetBuffer.set(sourceBuffer, targetStart);
```

Reply:
Thanks for the catch @devcat

Reply:
Resolved and added test, thanks @devcat